### PR TITLE
GEN ONE Eclipse_65 keyboard addition

### PR DESCRIPTION
**Pull request by rojasa1990**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by zvecr — CHANGES_REQUESTED** [review]
Same issue as #10644. You will have to fix up the added files as this PR cannot be merged with the files duplicated at the root level.

**Comment by rojasa1990** [comment]
> Same issue as #10644. You will have to fix up the added files as this PR cannot be merged with the files duplicated at the root level.

How do I fix the duplicate files from the root level problem?

**Comment by ridingqwerty** [comment]
> > Same issue as #10644. You will have to fix up the added files as this PR cannot be merged with the files duplicated at the root level.
> 
> How do I fix the duplicate files from the root level problem?

From the `qmk_firmware/` directory on your branch you should be able to do this:

`rm -rf genone`

`git status` will then show the following files being removed:

``` 
        deleted:    genone/eclipse_65/config.h
        deleted:    genone/eclipse_65/eclipse_65.c
        deleted:    genone/eclipse_65/eclipse_65.h
        deleted:    genone/eclipse_65/info.json
        deleted:    genone/eclipse_65/keymaps/default/keymap.c
        deleted:    genone/eclipse_65/readme.md
        deleted:    genone/eclipse_65/rules.mk
        deleted:    genone/genone_main.c
```

(note that `keyboards/genone/` files will still be intact)

Then you can do these to commit the changes to your branch:

```
git add .
git commit -m "Removing duplicate files"
git push -u origin master
```

**Comment by rojasa1990** [comment]
> > > Same issue as #10644. You will have to fix up the added files as this PR cannot be merged with the files duplicated at the root level.
> > 
> > 
> > How do I fix the duplicate files from the root level problem?
> 
> From the `qmk_firmware/` directory on your branch you should be able to do this:
> 
> `rm -rf genone`
> 
> `git status` will then show the following files being removed:
> 
> ```
>         deleted:    genone/eclipse_65/config.h
>         deleted:    genone/eclipse_65/eclipse_65.c
>         deleted:    genone/eclipse_65/eclipse_65.h
>         deleted:    genone/eclipse_65/info.json
>         deleted:    genone/eclipse_65/keymaps/default/keymap.c
>         deleted:    genone/eclipse_65/readme.md
>         deleted:    genone/eclipse_65/rules.mk
>         deleted:    genone/genone_main.c
> ```
> 
> (note that `keyboards/genone/` files will still be intact)
> 
> Then you can do these to commit the changes to your branch:
> 
> ```
> git add .
> git commit -m "Removing duplicate files"
> git push -u origin master
> ```

Where would I type these lines of code? I searched around my branch but did not find a place to write this.

**Comment by rojasa1990** [comment]
> > > > Same issue as #10644. You will have to fix up the added files as this PR cannot be merged with the files duplicated at the root level.
> > > 
> > > 
> > > How do I fix the duplicate files from the root level problem?
> > 
> > 
> > From the `qmk_firmware/` directory on your branch you should be able to do this:
> > `rm -rf genone`
> > `git status` will then show the following files being removed:
> > ```
> >         deleted:    genone/eclipse_65/config.h
> >         deleted:    genone/eclipse_65/eclipse_65.c
> >         deleted:    genone/eclipse_65/eclipse_65.h
> >         deleted:    genone/eclipse_65/info.json
> >         deleted:    genone/eclipse_65/keymaps/default/keymap.c
> >         deleted:    genone/eclipse_65/readme.md
> >         deleted:    genone/eclipse_65/rules.mk
> >         deleted:    genone/genone_main.c
> > ```
> > 
> > 
> > (note that `keyboards/genone/` files will still be intact)
> > Then you can do these to commit the changes to your branch:
> > ```
> > git add .
> > git commit -m "Removing duplicate files"
> > git push -u origin master
> > ```
> 
> Where would I type these lines of code? I searched around my branch but did not find a place to write this.

I figured it out on MSYS2

**Comment by rojasa1990** [comment]
> > > > > Same issue as #10644. You will have to fix up the added files as this PR cannot be merged with the files duplicated at the root level.
> > > > 
> > > > 
> > > > How do I fix the duplicate files from the root level problem?
> > > 
> > > 
> > > From the `qmk_firmware/` directory on your branch you should be able to do this:
> > > `rm -rf genone`
> > > `git status` will then show the following files being removed:
> > > ```
> > >         deleted:    genone/eclipse_65/config.h
> > >         deleted:    genone/eclipse_65/eclipse_65.c
> > >         deleted:    genone/eclipse_65/eclipse_65.h
> > >         deleted:    genone/eclipse_65/info.json
> > >         deleted:    genone/eclipse_65/keymaps/default/keymap.c
> > >         deleted:    genone/eclipse_65/readme.md
> > >         deleted:    genone/eclipse_65/rules.mk
> > >         deleted:    genone/genone_main.c
> > > ```
> > > 
> > > 
> > > (note that `keyboards/genone/` files will still be intact)
> > > Then you can do these to commit the changes to your branch:
> > > ```
> > > git add .
> > > git commit -m "Removing duplicate files"
> > > git push -u origin master
> > > ```
> > 
> > 
> > Where would I type these lines of code? I searched around my branch but did not find a place to write this.
> 
> I figured it out on MSYS2

Thanks

File: keyboards/genone/eclipse_65/config.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x0001 // GEN ONE LLC
```
Comment (drashna) [inline keyboards/genone/eclipse_65/config.h:22]: This VID is actually owned already.  Please change.

File: keyboards/genone/eclipse_65/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/genone/eclipse_65/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by rojasa1990** [comment]
I have added a GPL2+ compatible license header in my keymap.c file, and will update the file.

Also for the Vendor ID number, would that be purchased through a license or random? I have established my business with the IRS, State, and County.

________________________________

File: keyboards/genone/eclipse_65/config.h (hunk 1 above)
Comment (rojasa1990) [reply keyboards/genone/eclipse_65/config.h]: Thanks for the comment.

Updated with vendor ID 0xFEED and an available PID.

File: keyboards/genone/eclipse_65/keymaps/default/keymap.c (hunk 2 above)
Comment (rojasa1990) [reply keyboards/genone/eclipse_65/keymaps/default/keymap.c:17]: Thanks for the comment.
Added a GPL2+ license header in the keymap.c file.

**Review by drashna — APPROVED** [review]

File: keyboards/genone/eclipse_65/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	// Default layer
+	[0] = LAYOUT_all(
```
Comment (fauxpark) [inline keyboards/genone/eclipse_65/keymaps/default/keymap.c:22]:
```suggestion
	[0] = LAYOUT_65_ansi(
```

File: keyboards/genone/eclipse_65/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,          KC_HOME,
+		KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,         KC_PGUP,
+		KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                   KC_PGDN,
+		KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,                    KC_UP,  KC_END,
+		KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, MO(1), KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT),
+
+	// Fn1 Layer
+	[1] = LAYOUT_all(
```
Comment (fauxpark) [inline keyboards/genone/eclipse_65/keymaps/default/keymap.c:30]:
```suggestion
	[1] = LAYOUT_65_ansi(
```

File: keyboards/genone/eclipse_65/eclipse_65.h [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define ____ KC_NO
+
+
+#define LAYOUT_all( \
```
Comment (fauxpark) [inline keyboards/genone/eclipse_65/eclipse_65.h:24]:
```suggestion
#define LAYOUT_65_ansi( \
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/genone/eclipse_65/readme.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# GEN ONE Eclipse_65
+
+Eclipse_65 is a keyboard PCB supporting 65% layout. [More info at GEN ONE MERCH](https://genonemerch.com)
```
Comment (fauxpark) [inline keyboards/genone/eclipse_65/readme.md:3]: This link seems to be dead.

Comment (rojasa1990) [reply keyboards/genone/eclipse_65/readme.md:3]: I tried remaking the file using MSYS2 and received an error occurring with the keymap.c file. It says it has an error with the change from LAYOUT_ALL to LAYOUT_65_ANSI.

Comment (rojasa1990) [reply keyboards/genone/eclipse_65/readme.md:3]: # make genone/eclipse_65:default
QMK Firmware 0.9.36
WARNING:
 Can not run bin/qmk! This tool will be required when the develop branch is merged on 2020 Aug 29.

 Please run util/qmk_install.sh to install all the dependencies QMK requires.

Making genone/eclipse_65 with keymap default

avr-gcc.exe (GCC) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Size before:
   text    data     bss     dec     hex filename
      0   14414       0   14414    384e .build/genone_eclipse_65_default.hex

Compiling: keyboards/genone/eclipse_65/keymaps/default/keymap.c                                    keyboards/genone/eclipse_65/keymaps/default/keymap.c:22:8: error: implicit declaration of function 'LAYOUT_65_ansi'; did you mean 'LAYOUT_all'? [-Werror=implicit-function-declaration]
  [0] = LAYOUT_65_ansi(
        ^~~~~~~~~~~~~~
        LAYOUT_all
keyboards/genone/eclipse_65/keymaps/default/keymap.c:22:8: error: initializer element is not constant
keyboards/genone/eclipse_65/keymaps/default/keymap.c:22:8: note: (near initialization for 'keymaps[0][0][0]')
keyboards/genone/eclipse_65/keymaps/default/keymap.c:30:8: error: initializer element is not constant
  [1] = LAYOUT_65_ansi(
        ^~~~~~~~~~~~~~
keyboards/genone/eclipse_65/keymaps/default/keymap.c:30:8: note: (near initialization for 'keymaps[1][0][0]')
cc1.exe: all warnings being treated as errors
 [ERRORS]
 |
 |
 |
make[1]: *** [tmk_core/rules.mk:386: .build/obj_genone_eclipse_65_default/keyboards/genone/eclipse_65/keymaps/default/keymap.o] Error 1
Make finished with errors
make: *** [Makefile:584: genone/eclipse_65:default] Error 1

Comment (fauxpark) [reply keyboards/genone/eclipse_65/readme.md:3]: Well, CI is able to build it just fine so I think you are probably missing some of those changes locally.

Comment (rojasa1990) [reply keyboards/genone/eclipse_65/readme.md:3]: Thanks for the response. 
I changed all files that you requested me to change. What is missing from getting your approval?

Comment (fauxpark) [reply keyboards/genone/eclipse_65/readme.md:3]: You accepted all of my suggestions above, but did you then pull those changes down to your local clone from GitHub?

Other than the dead link here, everything looks good.

Comment (rojasa1990) [reply keyboards/genone/eclipse_65/readme.md:3]: I accepted and committed all suggestions you posted above. I manually removed main.c file and manually corrected the keymap.c file. When I open the files in my fork, they are all changed with you suggestions. 

Is the dead link a problem? how would I go about fixing that?

Comment (fauxpark) [reply keyboards/genone/eclipse_65/readme.md:3]: I get this when I go to that page:
image

Comment (rojasa1990) [reply keyboards/genone/eclipse_65/readme.md:3]: Ok, that's what the link is. We purchased the domain name and are currently working on the website. We are going to have it running next week.

**Review by fauxpark — APPROVED** [review]

File: keyboards/genone/eclipse_65/readme.md (hunk 6 above)
Comment (rojasa1990) [reply keyboards/genone/eclipse_65/readme.md:3]: How do I get ZVECR for another review for approval?

**Comment by rojasa1990** [comment]
How do I get ZVECR for another review for approval?

________________________________

**Comment by rojasa1990** [comment]
> > Same issue as #10644. You will have to fix up the added files as this PR cannot be merged with the files duplicated at the root level.
> 
> How do I fix the duplicate files from the root level problem?

I deleted duplicate files like requested.

**Comment by rojasa1990** [comment]
> Same issue as #10644. You will have to fix up the added files as this PR cannot be merged with the files duplicated at the root level.

Deleted duplicate files as requested.

**Comment by rojasa1990** [comment]
Thank You, so much.

________________________________

**Comment by rojasa1990** [comment]
Hello once again.

I made another pull request for my keyboard. "Added VIA folder under the keymaps folder #12021", for the Gen One Eclipse. This folder contains a keymap.c and rules.mk files to add VIA support.

Would you be available to review the pull request?

________________________________

**Pull request merged**